src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -188,23 +177,117 @@ function discoverController(
+    setAppState,
+    replaceUrlAppState,
+    isAppStateDirty,
+    kbnUrlStateStorage,
+    getPreviousAppState,
+  } = getState({
+    defaultAppState: getStateDefaults(),
+    storeInSessionStorage: config.get('state:storeInSessionStorage'),
+  });
+  if (appStateContainer.getState().index !== $scope.indexPattern.id) {
+    //used index pattern is different than the given by url/state which is invalid
+    setAppState({ index: $scope.indexPattern.id });
+  }
+  $scope.state = { ...appStateContainer.getState() };
+
+  // syncs `_g` portion of url with query services
+  const { stop: stopSyncingGlobalStateWithUrl } = syncQueryStateWithUrl(
+    data.query,
+    kbnUrlStateStorage
+  );
+
+  // sync initial app filters from state to filterManager
+  filterManager.setAppFilters(_.cloneDeep(appStateContainer.getState().filters));
+
+  const stopSyncingQueryAppStateWithStateContainer = connectToQueryState(
+    data.query,
+    appStateContainer,
+    { filters: esFilters.FilterStateStore.APP_STATE }
+  );
+
+  const appStateUnsubscribe = appStateContainer.subscribe(async newState => {
+    const { state: newStatePartial } = splitState(newState);
+    const { state: oldStatePartial } = splitState(getPreviousAppState());
+
+    if (!_.isEqual(newStatePartial, oldStatePartial)) {
+      $scope.$evalAsync(async () => {
+        $scope.state = { ...newState };
+
+        // detect changes that should trigger fetching of new data
+        const changes = ['interval', 'sort', 'index', 'query'].filter(
+          prop => !_.isEqual(newStatePartial[prop], oldStatePartial[prop])
+        );
+        if (changes.indexOf('index') !== -1) {
+          try {
+            $scope.indexPattern = await indexPatterns.get(newStatePartial.index);
+            $scope.opts.timefield = getTimeField();
+            $scope.enableTimeRangeSelector = !!$scope.opts.timefield;
+            // is needed to rerender the histogram
+            $scope.vis = undefined;
+
+            // Taking care of sort when switching index pattern:
+            // Old indexPattern: sort by A
+            // If A is not available in the new index pattern, sort has to be adapted and propagated to URL
+            const sort = getSortArray(newStatePartial.sort, $scope.indexPattern);
+            if (newStatePartial.sort && !_.isEqual(sort, newStatePartial.sort)) {
+              return await replaceUrlAppState({ sort });
+            }
+          } catch (e) {
+            toastNotifications.addWarning({ text: getIndexPatternWarning(newStatePartial.index) });
+          }
+        }
+
+        if (changes.length) {
+          $fetchObservable.next();
+        }
+      });
+    }
+  });
+
+  $scope.setIndexPattern = id => {
+    setAppState({ index: id });
+  };
+
+  // update data source when filters update
+  subscriptions.add(
+    subscribeWithScope($scope, filterManager.getUpdates$(), {
+      next: () => {
+        $scope.state.filters = filterManager.getAppFilters();
```

Comment:
This variable is not being used?

Reply:
it was used in the doc_table directive, but also not used in the directives code, so, good catch, removed it !